src/Compilers/CSharp/Test/Emit3/Semantics/PatternMatchingTests4.cs

```diff
@@ -4829,5 +4829,988 @@ class D : A { }
             Assert.Equal("x", x.ToString());
             Assert.Equal("A? x", model.GetDeclaredSymbol(x).ToTestDisplayString());
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75506")]
+        public void RedundantPattern_NotAOrBPattern()
+        {
+            var source = """
+object o = null;
+_ = o is not A or B; // 1
+_ = o is (not A) or B; // 2
```

Comment:
It might also be useful to test with interface types, both when they are unrelated, and when one inherits from the other. Since I think the subsumption behavior will be different than for unrelated class types. For example, if A and B are unrelated interfaces, then `not A or B` expresses: if it is an A, then it must also be a B, which is possible even when the declarations themselves are unrelated.